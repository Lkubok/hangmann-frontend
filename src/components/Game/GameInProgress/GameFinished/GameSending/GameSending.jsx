import React, { Component } from "react";
import * as selectors from "../../../../../reducers/selectors";
import { connect } from "react-redux";
import "./GameSending.scss";
import {
  clearGameParams,
  sendGameStat,
  fetchSingleQuote
} from "../../../../../actions/gameActions";
const { REACT_APP_API_HOST } = process.env;

export class GameSending extends Component {
  componentDidMount() {
    const { sendGameStat, scoreToSend, gameId, startTime } = this.props;
    const idToFetch = gameId.slice(0, 24);
    this.props.fetchSingleQuote(REACT_APP_API_HOST, idToFetch);
    const score = {
      ...scoreToSend,
      gameTime: `${Date.now() - startTime}`
    };
    sendGameStat(score);
  }
  render() {
    return <h6 className="database-info">game stored in database</h6>;
  }
}
const mapStateToProps = state => ({
  userName: selectors.getGameUserName(state),
  userEmail: selectors.getUserEmail(state),
  userLevel: selectors.getGameLevel(state),
  userLang: selectors.getGameLang(state),
  isGuessed: selectors.isGuessed(state),
  gameStatus: selectors.getStateOfGame(state),
  gameId: selectors.getGameId(state),
  searchedQuote: selectors.getSearchedQuote(state),
  scoreToSend: selectors.getScoreToSend(state),
  startTime: selectors.getStartTime(state)
});

const mapDispatchToProps = {
  clearGameParams,
  fetchSingleQuote,
  sendGameStat
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GameSending);
