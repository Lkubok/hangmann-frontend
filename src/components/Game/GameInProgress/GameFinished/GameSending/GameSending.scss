@import "../../../../../scss/mixins.scss";
@import "../../../../../scss/variables.scss";

.database-info {
  color: $dark-color;
  margin-bottom: 30px;
  font-weight: 200;
}
