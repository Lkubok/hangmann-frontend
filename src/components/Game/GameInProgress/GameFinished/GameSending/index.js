import GameSending from "./GameSending";
export default GameSending;
