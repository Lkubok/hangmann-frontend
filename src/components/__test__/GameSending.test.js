import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { GameSending } from "../Game/GameInProgress/GameFinished/GameSending/GameSending";
Enzyme.configure({ adapter: new Adapter() });

describe("GameSending component", () => {
  let props;
  const spyFunction = jest.fn();
  beforeEach(() => {
    props = {
      fetchSingleQuote: spyFunction,
      sendGameStat: spyFunction,
      scoreToSend: "win",
      gameId: "123",
      startTime: 123
    };
  });
  it("Should match snapshot", () => {
    const wrapper = shallow(<GameSending {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
