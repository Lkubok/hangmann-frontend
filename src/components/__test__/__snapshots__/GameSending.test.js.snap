// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameSending component Should match snapshot 1`] = `
<h6
  className="database-info"
>
  game stored in database
</h6>
`;
